File: chaerim/Programmers/Lv1_크레인_인형뽑기_게임.py
def solution(board, moves):
    answer = 0

    basket = []
    for move in moves:
        for _, y in enumerate(board):
            if y[move-1] != 0:
                basket.append(y[move-1])
                y[move-1] = 0
                break

    i = 0
    while i < len(basket)-1:
        if basket[i] == basket[i+1]:
            basket.pop(i)
            basket.pop(i)
            answer += 2
            i = 0
        else:
            i += 1
    
    return answer

File: sukjoo/Programmers/Lv1_크레인_인형뽑기_게임.java
public class 크레인_인형뽑기_게임 {

    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> stack = new Stack<>();
        for (int i = 0; i < moves.length; i++) {
            for (int j = 0; j < board.length; j++) {
                if (board[j][moves[i] - 1] > 0) {

                    if (!stack.isEmpty() && stack.peek() == board[j][moves[i] - 1]) {
                        answer += 2;
                        stack.pop();
                    } else {
                        stack.push(board[j][moves[i] - 1]);
                    }

                    board[j][moves[i] - 1] = 0;
                    break;

                }
            }
        }
        return answer;
    }